internal/templates/commands/render.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package render implements the "templates render" subcommand for installing a template.
+package commands
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/abcxyz/pkg/cli"
+)
+
+type Render struct {
+	cli.BaseCommand
+
+	fs fs
+
+	fv flagValues
+}
+
+type flagValues struct {
+	template       string
+	spec           string
+	dest           string
+	gitProtocol    string
+	logLevel       string
+	forceOverwrite bool
+	keepTempDirs   bool
+	inputs         map[string]string
+}
+
+// Desc implements cli.Command.
+func (r *Render) Desc() string {
+	return "Instantiate a template to setup a new app or add config files"
+}
+
+func (r *Render) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+  The {{ COMMAND }} command renders the given template into the given dest
+  directory.
+`
+}
+
+func (r *Render) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet()
+
+	sect := set.NewSection("Render options")
```

Comment:
Super nit: prefer `f` for this variable name for consistency with other CLIs.

Reply:
Done.